versioned_docs/version-V3/guides/providing-liquidity/decrease-liquidity.md

```diff
@@ -0,0 +1,68 @@
+---
+id: decrease-liquidity
+title: Decrease Liquidity
+sidebar_position: 4
+---
+
+Make sure to go through the [first guide](./setting-up-your-contract.md) before continuing to this section
+
+Here we decrease the liquidity of our position, without withdrawing all of it. This example assume the contract already has possession of the NFT, and requires that the calling address is the owner of that NFT.
+
+- In production, `amount0Min` and `amount1Min` should be adjusted to create slippage protections.
+
+## Decrease Liquidity
+
+```solidity
+    /// @notice A function that decreases the current liquidity by half. An example to show how to call the `decreaseLiquidity` function defined in periphery.
+    /// @param tokenId The id of the erc721 token
+    /// @return amount0 The amount received back in token0
+    /// @return amount1 The amount returned back in token1
+    function decreaseLiquidityInHalf(uint256 tokenId) external returns (uint256 amount0, uint256 amount1) {
+        // caller must be the owner of the NFT
+        require(msg.sender == deposits[tokenId].owner, 'Not the owner');
+        // get liquidity data for tokenId
+        uint128 liquidity = deposits[tokenId].liquidity;
+        uint128 halfLiquidity = liquidity / 2;
+
+        // amount0Min and amount1Min are price slippage checks
+        // if the amount received after burning is not greater than these minimums, transaction will fail
+        INonfungiblePositionManager.DecreaseLiquidityParams memory params =
+            INonfungiblePositionManager.DecreaseLiquidityParams({
+                tokenId: tokenId,
+                liquidity: halfLiquidity,
+                amount0Min: 0,
+                amount1Min: 0,
+                deadline: block.timestamp
+            });
+
+        (amount0, amount1) = nonfungiblePositionManager.decreaseLiquidity(params);
+
+        //send liquidity back to owner
+        _sendToOwner(tokenId, amount0, amount1);
+    }
+```
+
+## Sending Fees To The Calling Address
+
+This internal function takes the fees due amounts, which were previously populated in the last function, as arguments to `safeTransfer` which sends the fees which were previously collected into our contract to `owner`.
+
+```solidity
+    /// @notice Transfers funds to owner of NFT
+    /// @param tokenId The id of the erc721
+    /// @param amount0 The amount of token0
+    /// @param amount1 The amount of token1
+    function _sendToOwner(
```

Comment:
Why are we documenting _sendToOwner twice? Do you think thats better than just having a page on helper functions for this whole contract? Thought that this way might present this function as different from the _sendToOwner function that is documented before.

Reply:
Seems helpful to include helper functions, even redundantly, because users might come to the docs looking for only one of these liquidity features. trying to avoid fragmenting stuff as much as possible.